### ISSUE TITLE
PoC: Extension as a program

### ISSUE THREAD

**Issue by xlc** [open-web3-stack/XCQ#9]
Depends on #8 

Research if it is possible to implement an extension with a PolkaVM program.

The use case is that a chain can enable new extension by simply uploading a PolkaVM program and it will be used as extension implementation.

The PolkaVM program will be executed on a similar environment of the XCQ executor. i.e. it can call into other extensions. In that way, it could use an extension to read raw storages and basically do anything.

**Comment by indirection42** [comment]
Is this requirement an additional feature or expected results? I was thinking the implementation of the different extension are done in `host` side, and the caller provides polkavm blob (like smart contract) and other input data to do query.

**Comment by xlc** [comment]
This is an additional feature. Most of the extensions should still be implemented natively in runtime for performance reason. This will be useful to dynamically add new extensions without runtime upgrade and that can be useful in many use cases. However, this is a nice to have and not needed for the first release version.
